### PR TITLE
Adding an allow list configuration option

### PR DESCRIPTION
**Pull request by Ben-A-Canva**
## Overview
Adds an optional folder allowlist configuration.

## Motivation
Introduce an 'allowlist' feature to enable users to specify project folders explicitly, ensuring that only projects within these folders are included in the build process. This provides greater control over project compilation and improves workspace performance by preventing the accidental compilation of unrelated projects.

## Issue
With our monorepo structure, we are unable to use the extension because it attempts to spin up several hundred projects at once causing device OOM's and reboots. Adding an allowlist will help us to specifically target the projects we work on.

## Checklist

- [x] I have run this code and it appears to resolve the stated issue
- [x] `README.md` updated and added information about my change

### CODE REVIEW

File: src/manifest/dbtWorkspaceFolder.ts [hunk 1]
```diff
@@ -52,12 +52,25 @@ export class DBTWorkspaceFolder implements Disposable {
         `**/{${DBTProject.DBT_MODULES.join(",")}}`,
       ),
     );
-    const projectFiles = dbtProjectFiles
+    var projectFiles = dbtProjectFiles
```
Comment (mdesmet) [inline src/manifest/dbtWorkspaceFolder.ts:55]: Please never do `var`, rather keep on using `const` and apply the filter directly onto dbtProjectFiles inlining the same logic.

**Review by mdesmet — COMMENTED** [review]
Please also add a motivation on why this change is needed, ie what problems does it solve.

File: src/manifest/dbtWorkspaceFolder.ts (hunk 1 above)
Comment (Ben-A-Canva) [reply src/manifest/dbtWorkspaceFolder.ts]: yup, good callout. I'm not used to typescript 😁

**Comment by anandgupta42** [comment]
LGTM. @Ben-A-Canva, Can you please update the readme also?

**Review by mdesmet — APPROVED** [review]
LGTM, but please add docs as @anandgupta42 asked.

**Review by anandgupta42 — APPROVED** [review]

**Comment by anandgupta42** [comment]
LGTM @mdesmet the doc has been added. We can merge this.

**Comment by Ben-A-Canva** [comment]
> LGTM, but please add docs as @anandgupta42 asked.

Added an entry to the readme file. Is there anything else I need to update for docs? 👀

**Pull request merged**